logical/plugin/backend_server.go

```diff
@@ -0,0 +1,146 @@
+package plugin
+
+import (
+	"net/rpc"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/logical"
+)
+
+type backendPluginServer struct {
+	broker  *plugin.MuxBroker
+	backend logical.Backend
+}
+
+func (b *backendPluginServer) HandleRequest(args *HandleRequestArgs, reply *HandleRequestReply) error {
+	conn, err := b.broker.Dial(args.StorageID)
+	if err != nil {
+		*reply = HandleRequestReply{
+			Error: plugin.NewBasicError(err),
+		}
+		return nil
+	}
+	c := rpc.NewClient(conn)
+	defer c.Close()
+
+	storage := &StorageClient{client: c}
+	args.Request.Storage = storage
+
+	resp, err := b.backend.HandleRequest(args.Request)
+	*reply = HandleRequestReply{
+		Response: resp,
+		Error:    plugin.NewBasicError(err),
+	}
+
+	return nil
+}
+
+func (b *backendPluginServer) SpecialPaths(_ interface{}, reply *SpecialPathsReply) error {
+	*reply = SpecialPathsReply{
+		Paths: b.backend.SpecialPaths(),
+	}
+	return nil
+}
+
+func (b *backendPluginServer) Logger(_ interface{}, reply *BackendLoggerReply) error {
+	*reply = BackendLoggerReply{
+		Logger: b.backend.Logger(),
+	}
+	return nil
+}
+
+func (b *backendPluginServer) HandleExistenceCheck(args *HandleExistenceCheckArgs, reply *HandleExistenceCheckReply) error {
+	conn, err := b.broker.Dial(args.StorageID)
+	if err != nil {
+		*reply = HandleExistenceCheckReply{
+			Error: plugin.NewBasicError(err),
+		}
+		return nil
+	}
+	c := rpc.NewClient(conn)
+	defer c.Close()
+
+	storage := &StorageClient{client: c}
+	args.Request.Storage = storage
+
+	checkFound, exists, err := b.backend.HandleExistenceCheck(args.Request)
+	*reply = HandleExistenceCheckReply{
+		CheckFound: checkFound,
+		Exists:     exists,
+		Error:      plugin.NewBasicError(err),
+	}
+
+	return nil
+}
+
+func (b *backendPluginServer) Cleanup(_ interface{}, _ *struct{}) error {
+	b.backend.Cleanup()
+	return nil
+}
+
+func (b *backendPluginServer) Initialize(_ interface{}, _ *struct{}) error {
+	err := b.backend.Initialize()
+	return err
+}
+
+func (b *backendPluginServer) InvalidateKey(args string, _ *struct{}) error {
+	b.backend.InvalidateKey(args)
+	return nil
+}
+
+func (b *backendPluginServer) Configure(args *ConfigureArgs, reply *ConfigureReply) error {
```

Comment:
I put some comments related to this in earlier but if we're putting a function in the `logical.Backend` interface we shouldn't have a separate function in `framework.Backend`.

It seems there are two ways to go. One is to have the factory function take in config as normal and pass it to a factory function in the plugin which calls framework.Backend's Setup. This is assuming that plugins in fact use `framework` but they probably should. The other is to move it to the logical interface, which probably makes sense in that it is a common function.